pandas/core/groupby/generic.py

```diff
@@ -1569,6 +1574,220 @@ def func(df):
+        normalize: bool = False,
+        sort: bool = True,
+        ascending: bool = False,
+        dropna: bool = True,
+    ) -> DataFrame | Series:
+        """
+        Return a Series or DataFrame containing counts of unique rows.
+
+        .. versionadded:: 1.4.0
+
+        Parameters
+        ----------
+        subset : list-like, optional
+            Columns to use when counting unique combinations.
+        normalize : bool, default False
+            Return proportions rather than frequencies.
+        sort : bool, default True
+            Sort by frequencies.
+        ascending : bool, default False
+            Sort in ascending order.
+        dropna : bool, default True
+            Don’t include counts of rows that contain NA values.
+
+        Returns
+        -------
+        Series or DataFrame
+            Series if as_index is True, otherwise DataFrame.
+
+        See Also
+        --------
+        Series.value_counts: Equivalent method on Series.
+        DataFrame.value_counts: Equivalent method on DataFrame.
+        SeriesGroupBy.value_counts: Equivalent method on SeriesGroupBy.
+
+        Notes
+        -----
+        If the groupby as_index is True then the returned Series will have a
+        MultiIndex with one level per input column.
+        If the groupby as_index is False then the returned DataFrame will have an
+        additional column with the value_counts.
+        By default, rows that contain any NA values are omitted from
+        the result. By default, the result will be in descending order so that the
+        first element of each group is the most frequently-occurring row.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({
+        ...    'gender': ['male', 'male', 'female', 'male', 'female', 'male'],
+        ...    'education': ['low', 'medium', 'high', 'low', 'high', 'low'],
+        ...    'country': ['US', 'FR', 'US', 'FR', 'FR', 'FR']
+        ... })
+
+        >>> df
+            gender 	education 	country
+        0 	male 	low 	    US
+        1 	male 	medium 	    FR
+        2 	female 	high 	    US
+        3 	male 	low 	    FR
+        4 	female 	high 	    FR
+        5 	male 	low 	    FR
+
+        >>> df.groupby("gender").value_counts(normalize=True)
+        gender  education  country
+        female  high       FR         0.50
+                           US         0.50
+        male    low        FR         0.50
+                           US         0.25
+                medium     FR         0.25
+        Name: count, dtype: float64
+        """
+        RESULT_NAME = "count"
+        if self.axis == 1:
+            raise NotImplementedError(
+                "DataFrameGroupBy.value_counts only handles axis=0"
+            )
+
+        with self._group_selection_context():
+            df = self.obj
+
+            # Check for index rather than column grouping
+            index_grouping = self.grouper.groupings[0].name is None
```

Comment:
Is this for handling the case of `df.groupby(df.index)`? If so, I don't believe this is reliable. E.g.

```
df = pd.DataFrame({'a': [1, 2], 'b': [3, 4]})
df.index.name = 'myname'
print(df.groupby(df.index).grouper.groupings[0].name)
```

gives "myname"

Reply:
Also, is this hit by the tests?

Reply:
Thanks for pointing that out. I have found a better way, but it is still painful to detect what `get_grouper()` decided. My `test_against_frame_and_seriesgroupby()` tests with by=array and by=function.